web/assets/js/components/gamecontrols.ts

```diff
@@ -1,185 +1,206 @@
-      this.validationTL.play();
-    } else if (this.action === action && direct) {
-      if (this.pendingDraw) {
-        // if pending draw refused
-        this._declineOffer();
-      }
-      this.action = "void";
-      this[action === "resign" ? "disableCtr1TL" : "disableCtr0TL"].reverse();
-      this[action === "resign" ? "swithCtr0TL" : "swithCtr1TL"].reverse();
-
-      this.validationTL.reverse().then(() => {
-        this.DOM.timer.innerHTML = this.timer;
-        gsap.set(this.DOM.timer, { autoAlpha: 0, display: "none" });
-      });
+    constructor(opts: any) {
+        super(opts);
+        this.events = {} as Events;
     }
-  }
 
-  async _clickOnConfirm() {
-    if (this.action === "resign") {
-      Action.requestResign();
-      this.call("goTo", ["/"], "router");
+    init() {
+        // automatically called at start
+        console.log("PlayControls component init");
+
+        //Vars
+        this.action = "void" as GameAction;
+        this.pendingDraw = false;
+        this.timer = 9;
+
+        this.contents = {
+            resign: {
+                title: "Resign",
+                content: "Do you really want to Resign the Match?",
+                btn: "Confirm",
+            },
+            draw: {
+                title: "Draw",
+                content: "Do you really want to offer a draw?",
+                btn: "Confirm",
+            },
+            offer: {
+                title: "Draw",
+                content: "You were offered a Draw. Do you accept?",
+                btn: "Agree",
+            },
+        };
+
+        //DOM
+        this.DOM.ctr0 = this.DOM.el.querySelector("#js-gamecontrols-ctr0");
+        this.DOM.ctr1 = this.DOM.el.querySelector("#js-gamecontrols-ctr1");
+        this.DOM.ctrConfirm = this.DOM.el.querySelector("#js-gamecontrols-confirm");
+        this.DOM.paneValidation = this.DOM.el.querySelector("#js-validation");
+        this.DOM.title = this.DOM.el.querySelector(".js-gamecontrols-title");
+        this.DOM.content = this.DOM.el.querySelector(".js-gamecontrols-content");
+        this.DOM.timer = this.DOM.el.querySelector("#js-gamecontrols-confirm-timer");
+        this.DOM.ctrConfirmContent = this.DOM.el.querySelector("#js-gamecontrols-confirm-content");
+
+        //controls events
+        this.events.clickOnCtr0 = this.on({
+            e: "click",
+            target: this.DOM.ctr0,
+            cb: this._clickOnCtr.bind(this, "resign", true),
+        });
+        this.events.clickOnCtr1 = this.on({
+            e: "click",
+            target: this.DOM.ctr1,
+            cb: this._clickOnCtr.bind(this, "draw", true),
+        });
+        this.events.clickOnConfirm = this.on({
+            e: "click",
+            target: this.DOM.ctrConfirm,
+            cb: this._clickOnConfirm.bind(this),
+        });
+
+        //async events
+        Events.on("drawPropal", this._getDrawProposition.bind(this));
```

Comment:
The Event listener here must be replaced by the long polling method or the emitter should be reimplemented in the new Action `listenDraw` polling method

Reply:
See here:
https://github.com/gnolang/gnochess/pull/21/files#r1329978134